sound/soc/sof/pm.c

```diff
@@ -213,6 +217,9 @@ static int sof_suspend(struct device *dev, bool runtime_suspend)
 	/* release trace */
 	snd_sof_release_trace(sdev);
 
+	/* Notify clients about core suspend */
+	sof_suspend_clients(sdev, pm_state);
```

Comment:
I'm wondering why this is needed? All SOF clients have the main SOF device as their parent, right? So, when the parent is suspended, all the children should have their `.suspend()` callbacks called too? Isn't that happening, and if not, maybe there's a problem somewhere?

Reply:
I think the system suspend will be but the runtime is not.
If we want the runtime callbacks to be called then the driver would need to implement the pm_runtime_* handling itself, but that will not work with the dma-trace.
We don't want to block the parent's runtime suspend by keeping the child enabled but if we disable the child to let the parent runtime suspend then the child will be not runtime resume when along with the core as it is marked as disabled, there must be something which tells the driver that it's driver can call pm_runtime_get_sync() to get it's runtime resume to be called.
Or something....

Reply:
Sorry @ujfalusi what's so special about the dma trace?

I don't follow what you are trying to say.

On my side I assumed that somehow the trace IPC would be forwarded to the dma trace client, and that you can use this information to call pm_runtime_get_sync(), and use put_sync() when the interrupt and trace management are handled. '

Or not handle pm_runtime at all, that would not prevent the parent from dealing with pm_runtime, would it?

Reply:
> Sorry @ujfalusi what's so special about the dma trace?
> 
> I don't follow what you are trying to say.
> 
> On my side I assumed that somehow the trace IPC would be forwarded to the dma trace client, and that you can use this information to call pm_runtime_get_sync(), and use put_sync() when the interrupt and trace management are handled. '

But the trace IPC message will be not sent unless we enable it prior, but we don't know when the DSP is resumed.
When the DSP is resumed we need to re-enable the DMA trace and prior when it is going to be supended we need to disable it. There is nothing going to come when the DSP is off, but we must not keep it on.

> Or not handle pm_runtime at all, that would not prevent the parent from dealing with pm_runtime, would it?

I have updated the patch and the dma-trace now does a pm_runtime_no_callbacks() to it's device.